warehouse/core/src/main/java/datawave/iterators/filter/ColumnVisibilityAndFilter.java

```diff
@@ -6,19 +6,21 @@
 import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Value;
 
+import java.util.Arrays;
+
 /**
  * This subclass of the {@code RegexFilterBase} class is used to filter based on the column visibility of the {@code Key} object {@code k}.
  */
 public class ColumnVisibilityAndFilter extends TokenFilterBase {
     
     /**
-     * This method returns a {@code Iterator<String>} iterator of tokens containing the column column visibility parts for {@code Key} object {@code k}
-     * 
+     * Determine if the key's column visibility matches the regex defined by {@link AgeOffConfigParams#MATCHPATTERN}
+     *
      * @param k
      *            {@code Key} object containing the row, column family, and column qualifier.
      * @param v
      *            {@code Value} object containing the value corresponding to the {@code Key: k}
-     * @return {@code String} object containing the column visibility of {@code k}
+     * @return true if the provided key's column visibility matches the configured {@link TokenFilterBase}'s pattern
      */
     @Override
     public boolean hasToken(Key k, Value v, byte[][] testTokens) {
```

Comment:
most of this method can be moved to an abstract method in TokenFilterBase

Reply:
done